python/example/brunel.py

```diff
@@ -0,0 +1,161 @@
+    parser.add_argument('-e', '--n-external', dest='next', type=int, default=40, help='Number of incoming Poisson (external) connections per cell')
+    parser.add_argument('-p', '--in-degree-prop', dest='syn_per_cell_prop', type=float, default=0.05, help='Proportion of the connections received per cell')
+    parser.add_argument('-w', '--weight', dest='weight', type=float, default=1.2, help='Weight of excitatory connections')
+    parser.add_argument('-d', '--delay', dest='delay', type=float, default=0.1, help='Delay of all connections')
+    parser.add_argument('-g', '--rel-inh-w', dest='rel_inh_strength', type=float, default=1, help='Relative strength of inhibitory synapses with respect to the excitatory ones')
+    parser.add_argument('-l', '--lambda', dest='poiss_lambda', type=float, default=1, help='Expected number of spikes from a single poisson cell (kHz)')
+    parser.add_argument('-t', '--tfinal', dest='tfinal', type=float, default=100, help='Length of the simulation period (ms)')
+    parser.add_argument('-s', '--dt', dest='dt', type=float, default=1, help='Simulation time step (ms)')
+    parser.add_argument('-G', '--group-size', dest='group_size', type=int, default=10, help='Number of cells per cell group')
+    parser.add_argument('-S', '--seed', dest='seed', type=int, default=42, help='Seed for poisson spike generators')
+    parser.add_argument('-f', '--write-spikes', dest='spike_file_output', action='store_true', help='Save spikes to file')
+    parser.add_argument('-z', '--profile-rank-zero', dest='profile_only_zero', action='store_true', help='Only output profile information for rank 0')
+    parser.add_argument('-v', '--verbose', dest='verbose', action='store_true', help='Print more verbose information to stdout')
+
+    for k,v in vars(parser.parse_args()).items():
+        setattr(opt, k, v)
+
+    context = arbor.context()
+    print(context)
+
+    meters = arbor.meter_manager()
+    meters.start(context)
+
+    recipe = brunel_recipe(opt.nexc, opt.ninh, opt.next, opt.syn_per_cell_prop, opt.weight, opt.delay, opt.rel_inh_strength, opt.poiss_lambda, opt.seed)
+
+    meters.checkpoint('recipe-create', context)
+
+    hint = arbor.partition_hint()
+    hint.cpu_group_size = opt.group_size
+    hints = {arbor.cell_kind.lif: hint}
+    decomp = arbor.partition_load_balance(recipe, context, hints)
+    print(decomp)
+
+    meters.checkpoint('load-balance', context)
+
+    sim = arbor.simulation(recipe, decomp, context)
+    sim.record(arbor.spike_recording.all)
+
+    meters.checkpoint('simulation-init', context)
+
+    sim.run(opt.tfinal,opt.dt)
+    print('Simulation finished')
+
+    meters.checkpoint('simulation-run', context)
+
+    # Print profiling information
+    print(f'{arbor.meter_report(meters, context)}')
+
+    # Print spike times
+    print(f"{len(sim.spikes())} spikes generated.")
+
+    if opt.spike_file_output:
+        with open('spikes.txt', 'w') as the_file:
```

Comment:
The C++ example writes to `spikes.gdf` instead. Ideally, neither this example nor the C++ one would write to a hardcoded path, but instead use a path supplied as an argument to `-f`.